tools/fast_build_tar/buildtar.go

```diff
@@ -0,0 +1,531 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// fast tar builder for bazel
+package main
+
+import (
+	"archive/tar"
+	"bufio"
+	"compress/bzip2"
+	"compress/gzip"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+
+	"golang.org/x/build/pargzip"
+)
+
+type tarFile struct {
+	directory string
```

Comment:
what is this used for?

Reply:
it's the directory in which to store the file inside the layer